File: iteration1/src/Person.java
public abstract class Person {
    // Attributes

    private String name;
    private String surname;
    private String email;
    private String phoneNumber; 
    private String ID;
    private String password;
    private String faculty;
    private String department;

    // Constructor

    public Person (String name, String surname, String email, String phoneNumber, String ID, String password, String faculty, String department) {
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.ID = ID;
        this.password = password;
        this.faculty = faculty;
        this.department = department;
    }

    // Getters

    public String getName() {
        return name;
    }

    public String getSurname() {
        return surname;
    }

    public String getEmail() {
        return email;
    }
    
    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getID() {
        return ID;
    }

    public String getPassword() {
        return password;
    }

    public String getFaculty() {
        return faculty;
    }

    public String getDepartment() {
        return department;
    }

    // Setters

    public void setName(String name) {
        this.name = name;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public void setID(String iD) {
        ID = iD;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setFaculty(String faculty) {
        this.faculty = faculty;
    }

    public void setDepartment(String department) {
        this.department = department;
    }

    // Abstract methods

    public int getSemester() {
        return 0;
    }

    public Arraylist<Course> getCoursesTaken() {
        return null;
    }

    public Transcript getTranscript() {
        return null;
    }

    public Advisor getAdvisor() {
        return null;
    }

    public ArrayList<Student> getStudents() {
        return null;
    }
    
    public void setSemester(int semester) {
    }

    public void setCoursesTaken(ArrayList<Course> coursesTaken) {
    }

    public void setTranscript(Transcript transcript) {
    }

    public void setAdvisor(Advisor advisor) {
    }

    public void setStudents(ArrayList<Student> students) {
    }

}

File: iteration3/src/person.py
from abc import ABC, abstractmethod


class Person(ABC):
    def __init__(
        self,
        name: str,
        surname: str,
        email: str,
        phone_number: str,
        id: str,
        password: str,
        faculty: str,
        department: str,
    ):
        self.__name = name
        self.__surname = surname
        self.__email = email
        self.__phone_number = phone_number
        self.__id = id
        self.__password = password
        self.__faculty = faculty
        self.__department = department

    def get_name(self) -> str:
        return self.__name

    def get_surname(self) -> str:
        return self.__surname

    def get_email(self) -> str:
        return self.__email

    def get_phone_number(self) -> str:
        return self.__phone_number

    def get_id(self) -> str:
        return self.__id

    def get_password(self) -> str:
        return self.__password

    def get_faculty(self) -> str:
        return self.__faculty

    def get_department(self) -> str:
        return self.__department

    def set_name(self, name):
        self.__name = name

    def set_surname(self, surname):
        self.__surname = surname

    def set_email(self, email):
        self.__email = email

    def set_phone_number(self, phone_number):
        self.__phone_number = phone_number

    def set_id(self, id):
        self.__id = id

    def set_password(self, password):
        self.__password = password

    def set_faculty(self, faculty):
        self.__faculty = faculty

    def set_department(self, department):
        self.__department = department
